Source/RxSwift/distinct.swift

```diff
@@ -19,12 +19,12 @@ extension Observable {
      */
     public func distinct(_ predicate: @escaping (Element) throws -> Bool) -> Observable<Element> {
         var cache = [Element]()
-        return flatMap { element -> Observable<Element> in
+        return compactMap { element -> Element? in
             if try cache.contains(where: predicate) {
-                return Observable<Element>.empty()
+                return nil
             } else {
                 cache.append(element)
-                return Observable<Element>.just(element)
+                return element
```

Comment:
Wondering if it's worth using a set and filter similarly to what we've done in CombineExt. WDYT?
https://github.com/CombineCommunity/CombineExt/blob/master/Sources/Operators/RemoveAllDuplicates.swift#L22-L24

Reply:
I agree with you